pkg/networking/networking.go

```diff
@@ -61,45 +61,35 @@ func (crt *customRoundTripper) RoundTrip(request *http.Request) (*http.Response,
 func NewNetworkAccess(config configuration.Configuration) NetworkAccess {
 	// prepare logger
 	logger := log.New(os.Stderr, "NetworkAccess - ", config.GetInt(configuration.DEBUG_FORMAT))
-	if config.GetBool(configuration.DEBUG) == false {
+	if !config.GetBool(configuration.DEBUG) {
 		logger.SetOutput(io.Discard)
 	}
 
-	c := networkImpl{
+	return &networkImpl{
 		config:       config,
 		staticHeader: http.Header{},
 		logger:       logger,
 		proxy:        http.ProxyFromEnvironment,
 	}
-
-	return &c
 }
 
-func (n *networkImpl) AddHeaderField(key string, value string) {
-	n.staticHeader[key] = append(n.staticHeader[key], value)
+func (n *networkImpl) AddHeaderField(key, value string) {
+	n.staticHeader.Add(key, value)
 }
 
 func (n *networkImpl) AddHeaders(request *http.Request) error {
 	n.addDefaultHeader(request)
-	err := middleware.AddAuthenticationHeader(n.GetAuthenticator(), n.config, request)
-	return err
+	return middleware.AddAuthenticationHeader(n.GetAuthenticator(), n.config, request)
 }
 
 // addDefaultHeader adds the default headers request.
 func (n *networkImpl) addDefaultHeader(request *http.Request) {
-	defaultHeader := http.Header{}
-
-	// add static header
-	for k, v := range n.staticHeader {
-		for i := range v {
-			defaultHeader.Add(k, v[i])
-		}
-	}
-
 	// iterate over default headers and add them if there is no existing entry yet
-	for k, v := range defaultHeader {
-		for i := range v {
-			request.Header.Set(k, v[i])
+	for k, v := range n.staticHeader.Clone() {
+		if request.Header.Values(k) == nil {
+			for i := range v {
+				request.Header.Add(k, v[i])
+			}
```

Comment:
I believe this logic better aligns with the comment above. Though, the comment may be out-of date so please take a look.

Reply:
Yes, the comment is actually out of date. We changed it here to actually use set, so that the framework overrides header that might already be in the request. One example are headers that are set by the legacy CLI and need to be replaced by the Extensible CLI.

Reply:
Cool, then I think the key must be deleted if the static headers specify it, and then add header values to be correct.

Reply:
Question: Is there a reason to use del() and add() instead of set()?

Reply:
Yes, you can have multiple header values for a same header, for instance: `Cookies` or `Set-Cookies` is an example.
If you happen to have multiple `Cookies` headers in your `staticHeader` object, only the last `Cookie` header would remain using the `Set` method.